Index: dojo/dom-construct.js
===================================================================
--- dojo/dom-construct.js
+++ dojo/dom-construct.js
@@ -96,11 +96,11 @@
 		if(!masterId){
 			doc[masterName] = masterId = ++masterNum + "";
 			masterNode[masterId] = doc.createElement("div");
-		}
-
-		if(has("ie") <= 8){
-			if(!doc.__dojo_html5_tested && doc.body){
+			
+			if(has("ie") <= 8){
+				doc = doc.createDocumentFragment();
 				html5domfix(doc);
+				doc.appendChild(masterNode[masterId]);
 			}
 		}
 
